### PR TITLE
Moved the Argument Parser code into its own module. Closes issue #15.

### PR DESCRIPTION
**Pull request by flamesjames**
Please let me know if there's any modifications that should be made or if I misunderstood which code should be contained in the module. 

I also corrected some reference issues with some of the imports that used "." notation given that was giving me issues.

Closes issue #15

### CODE REVIEW

File: src/gcp_scanner/argparser.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import argparse
```
Comment (mshudrak) [inline src/gcp_scanner/argparser.py:1]: Please run pylint in order to check for Python style violations.

File: src/gcp_scanner/scanner.py [hunk 2]
```diff
@@ -24,14 +24,15 @@
 import sys
 from typing import List, Tuple, Dict, Optional
 
-from . import crawl
-from . import credsdb
+import crawl
```
Comment (mshudrak) [inline src/gcp_scanner/scanner.py:27]: This is required in order to use the tool as a pip module.

File: src/gcp_scanner/scanner.py [hunk 3]
```diff
@@ -286,78 +287,9 @@ def main():
...
-  parser.add_argument(
-      '-l',
-      '--logging',
-      default='WARNING',
-      dest='log_level',
-      choices=('INFO', 'WARNING', 'ERROR'),
-      help='Set logging level (INFO, WARNING, ERROR)')
-
-  args = parser.parse_args()
+  # Call ArgParser module to parse arguments
```
Comment (mshudrak) [inline src/gcp_scanner/scanner.py:290]: I think the function call is self explanatory

File: src/gcp_scanner/scanner.py [hunk 4]
```diff
@@ -286,78 +287,9 @@ def main():
...
+  args = argparser.parse_arguments()
+
   if not args.key_path and not args.gcloud_profile_path \
```
Comment (mshudrak) [inline src/gcp_scanner/scanner.py:293]: I think the code below is also a good candidate to be moved into a separate module/function.

**Review by mshudrak — CHANGES_REQUESTED** [review]
Thanks for submitting PR, PTAL at my comments.

**Comment by flamesjames** [comment]
@mshudrak - sure thing, I will make these updates tomorrow. Thank you for the feedback

File: src/gcp_scanner/argparser.py (hunk 1 above)
Comment (flamesjames) [reply src/gcp_scanner/argparser.py:1]: Have the updates ready in upcoming commit `49ca810`

File: src/gcp_scanner/scanner.py (hunk 3 above)
Comment (flamesjames) [reply src/gcp_scanner/scanner.py:290]: This has been removed with upcoming commit.

File: src/gcp_scanner/scanner.py (hunk 2 above)
Comment (flamesjames) [reply src/gcp_scanner/scanner.py:27]: These have been corrected with upcoming commit.

File: src/gcp_scanner/scanner.py (hunk 4 above)
Comment (flamesjames) [reply src/gcp_scanner/scanner.py:293]: @mshudrak - are you referring to just lines 291-297 or all of the code in the rest of the function (including or excluding the crawl_loop func call)?

Comment (mshudrak) [reply src/gcp_scanner/scanner.py:293]: Yes, I refer to lines 361-367 in the original file. and I think 293-299 in the new one. Basically the if statement where we print 

```   
 logging.error(
        'Please select at least one option to begin scan\
 -k/--sa_key_path,-g/--gcloud_profile_path, -m, -rt, -at'
    )
```

Comment (flamesjames) [reply src/gcp_scanner/scanner.py:293]: Got it

**Comment by flamesjames** [comment]
@mshudrak - I have made the suggested modifications. Please let me know if anything else needs addressed. Thanks!

File: README.md [hunk 5]
```diff
@@ -42,6 +42,15 @@ To install the package, use `pip` (you must also have `git` installed):
 
 ```
 pip install https://github.com/google/gcp_scanner/archive/main.zip
+gcp-scanner --help
```
Comment (mshudrak) [inline README.md:45]: Please use rebase to maintain clean commit history.

File: src/gcp_scanner/scanner.py [hunk 6]
```diff
@@ -26,12 +25,14 @@
...
 from . import credsdb
+from . import argparser
 from google.cloud import container_v1
 from google.cloud import iam_credentials
 from google.cloud.iam_credentials_v1.services.iam_credentials.client import IAMCredentialsClient
 from googleapiclient import discovery
 from httplib2 import Credentials
-from .models import SpiderContext
+from models import SpiderContext
+import argparser
```
Comment (mshudrak) [inline src/gcp_scanner/scanner.py:35]: argparser is imported twice

File: src/gcp_scanner/scanner.py [hunk 7]
```diff
@@ -26,12 +25,14 @@
...
 from . import crawl
 from . import credsdb
+from . import argparser
 from google.cloud import container_v1
 from google.cloud import iam_credentials
 from google.cloud.iam_credentials_v1.services.iam_credentials.client import IAMCredentialsClient
 from googleapiclient import discovery
 from httplib2 import Credentials
-from .models import SpiderContext
+from models import SpiderContext
```
Comment (mshudrak) [inline src/gcp_scanner/scanner.py:34]: I don't think you changed SpiderContext, why is this changed?

**Review by mshudrak — CHANGES_REQUESTED** [review]
Hi, thanks for the fix. I left some comments. PTAL.

File: README.md (hunk 5 above)
Comment (flamesjames) [reply README.md:45]: @mshudrak apologies, I am fairly new to git and open source. Where should I rebase back to?

Comment (mshudrak) [reply README.md:45]: I'd suggest to check this blog https://www.atlassian.com/git/tutorials/merging-vs-rebasing. Basically, your commits need to go on top of the one where I introduced this change.

**Comment by flamesjames** [comment]
@mshudrak - Hello, I made some updates. I tried to rebase my branch/project to the best of my ability but not sure if I did it right. That was my first time rebasing ever.. I think I got my branch up-to-date though and made your suggested changes. Please let me know if there are any more modifications and thank you for being patient with me as I'm learning how this whole process works.

File: src/gcp_scanner/argparser.py (hunk 1 above)
Comment (flamesjames) [reply src/gcp_scanner/argparser.py:2]: @mshudrak - sorry about this.. it's weird, I ran the pylint package locally on that file and it's telling me different spacing warnings.. taking the GitHub Actions build test comments into consideration, I hope the file is okay with the most recent commits...apologies if there are still tweaks to be made

**Comment by mshudrak** [comment]
Sorry @flamesjames but I don't see you addressed all my suggestions.

File: src/gcp_scanner/scanner.py (hunk 7 above)
Comment (flamesjames) [reply src/gcp_scanner/scanner.py:34]: Corrected.

File: src/gcp_scanner/scanner.py (hunk 6 above)
Comment (flamesjames) [reply src/gcp_scanner/scanner.py:35]: Corrected.

**Comment by flamesjames** [comment]
> Sorry @flamesjames but I don't see you addressed all my suggestions.

@mshudrak Apologies - the rebase messed up some of my changes.. pushed the updates for 2 package issues..

File: README.md (hunk 5 above)
Comment (mshudrak) [reply README.md:45]: This change is still visible.

File: src/gcp_scanner/argparser.py [hunk 8]
```diff
@@ -0,0 +1,96 @@
+"""This Module can be imported to parse command line arguments."""
+import argparse
+import logging
+
+def parse_arguments():
```
Comment (mshudrak) [inline src/gcp_scanner/argparser.py:5]: Please follow the style of function description and arguments description that we use in all other functions.

**Review by mshudrak — CHANGES_REQUESTED** [review]
Few more comments to address. Thanks.

File: src/gcp_scanner/argparser.py (hunk 8 above)
Comment (flamesjames) [reply src/gcp_scanner/argparser.py:5]: @mshudrak - ok sorry, I will add that, but can you please help me figure out how to remove the Readme.md change? I'm new to GitHub and not sure what happened or how to get that out of my branch - I checked the rebase stuff and thought I corrected it but guess not. Any help is much appreciated.

Comment (mshudrak) [reply src/gcp_scanner/argparser.py:5]: Let's just address other comments first. I will help you with the README.md changes once you fix them.

**Review by mshudrak — CHANGES_REQUESTED** [review]
Let's fix my other comments. I will help with README.md once we are done with other issues.

**Comment by mshudrak** [comment]
It is not conflicting with scanner.py. You can resolve conflict or it might be just easier at this point to close this one and start another one with the fresh version of the scanner.

**Comment by mshudrak** [comment]
It was resolved by another PR #93 . Closing this PR.
